src/include/sof_api/ipc4/base-config.h

```diff
@@ -0,0 +1,240 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
```

Comment:
So I agree having an API directory for modules is good, but can we call it include/module/ ? This way all our modules would be including e.g.
```c
#include <module/ipc4/base-config.h>
#include <module/module.h>
```
and so on. i.e. we make it clear this API is for modules and make it more obvious when new APIs are added if they are intended for baseFW or are intened for client modules.

Reply:
Idea behind it is to keep this files as an external header package and calling it sof_api doesn't leave place for mistake.

Reply:
Modules include are limited only to include/sof_api directory, so its include will looks like:
`#include <ipc4/base-config.h>
#include <module/generic.h>`

From the sof perspective it is:
`#include <sof_api/ipc4/base-config.h>
#include <sof_api/module/generic.h>`

Reply:
> Modules include are limited only to include/sof_api directory, so its include will looks like:

I'm good with that. My issue is that ```sof_api`` name is quite ambiguous i.e. API for SOF what ? all of SOF or just part of SOF. In this case it needs to be clear that this API is only for modules only.

Reply:
These headers will be kept in this repo for CI needs. If any changes will be required we will create headers release package for loadable modules and change api version number to keep on track changes. For that will have to create some kind of ABI script only for this cathalog.

Reply:
> > Modules include are limited only to include/sof_api directory, so its include will looks like:
> 
> I'm good with that. My issue is that `sof_api` name is quite ambiguous i.e. API for SOF what ? all of SOF or just part of SOF. In this case it needs to be clear that this API is only for modules only.

sof_api because we are using module_api, component_api and that headers contains essential data structures for modules to be loaded. If naming is such a problem we can move them to lmdk/include

Reply:
The direction of travel is component api is deprecated and will be removed, we will only have module api for modules so this programming surface must be obvious for module developers. In the future we will have the following include directories (not showing IADK atm)
1) src/include/module - all APIs used for modules
2) src/include/sof - base FW apis, not used by modules.
3) src/audio/<module_name>/ - local module private headers, used by module only.

Reply:
@lgirdwood I will rename `sof_api` to `module` soon.